client/src/main/com/sinch/sdk/models/Configuration.java

```diff
@@ -488,6 +523,63 @@ public Builder setConversationContext(ConversationContext context) {
       return this;
     }
 
+    /**
+     * Set Mailgun related api key
+     *
+     * @param apiKey {@link MailgunCredentials#getApiKey()}
+     * @return Current builder
+     * @since __TO_BE_DEFINED__
+     */
+    public Builder setMailgunApiKey(String apiKey) {
+      if (null == this.mailgunContext) {
```

Comment:
check `this.mailgunCredentials` instead

Reply:
Right, seems a far better test ! 
Thank you.
Fixed